api/db-cdr/generate-cdr/make-bq-denormalized-search.sh

```diff
@@ -158,53 +183,102 @@ WHERE x.person_id = y.person_id"
 # set has EHR data flag
 ################################################
 echo "set has EHR data flag in cb_search_person"
-bq --quiet --project=$BQ_PROJECT query --nouse_legacy_sql \
-"UPDATE \`$BQ_PROJECT.$BQ_DATASET.cb_search_person\` x
-SET x.has_ehr_data = y.has_data
-FROM
-    (
-        SELECT a.person_id, CASE WHEN b.person_id is not null THEN 1 ELSE 0 END AS has_data
-        FROM \`$BQ_PROJECT.$BQ_DATASET.person\` a
-        LEFT JOIN
-            (
-                SELECT DISTINCT person_id
-                FROM \`$BQ_PROJECT.$BQ_DATASET.measurement\` as a
-                LEFT JOIN \`$BQ_PROJECT.$BQ_DATASET.measurement_ext\` as b on a.measurement_id = b.measurement_id
-                WHERE lower(b.src_id) like 'ehr site%'
-                UNION DISTINCT
-                SELECT DISTINCT person_id
-                FROM \`$BQ_PROJECT.$BQ_DATASET.condition_occurrence\` as a
-                LEFT JOIN \`$BQ_PROJECT.$BQ_DATASET.condition_occurrence_ext\` as b on a.condition_occurrence_id = b.condition_occurrence_id
-                WHERE lower(b.src_id) like 'ehr site%'
-                UNION DISTINCT
-                SELECT DISTINCT person_id
-                FROM \`$BQ_PROJECT.$BQ_DATASET.device_exposure\` as a
-                LEFT JOIN \`$BQ_PROJECT.$BQ_DATASET.device_exposure_ext\` as b on a.device_exposure_id = b.device_exposure_id
-                WHERE lower(b.src_id) like 'ehr site%'
-                UNION DISTINCT
-                SELECT DISTINCT person_id
-                FROM \`$BQ_PROJECT.$BQ_DATASET.drug_exposure\` as a
-                LEFT JOIN \`$BQ_PROJECT.$BQ_DATASET.drug_exposure_ext\` as b on a.drug_exposure_id = b.drug_exposure_id
-                WHERE lower(b.src_id) like 'ehr site%'
-                UNION DISTINCT
-                SELECT DISTINCT person_id
-                FROM \`$BQ_PROJECT.$BQ_DATASET.observation\` as a
-                LEFT JOIN \`$BQ_PROJECT.$BQ_DATASET.observation_ext\` as b on a.observation_id = b.observation_id
-                WHERE lower(b.src_id) like 'ehr site%'
-                UNION DISTINCT
-                SELECT DISTINCT person_id
-                FROM \`$BQ_PROJECT.$BQ_DATASET.procedure_occurrence\` as a
-                LEFT JOIN \`$BQ_PROJECT.$BQ_DATASET.procedure_occurrence_ext\` as b on a.procedure_occurrence_id = b.procedure_occurrence_id
-                WHERE lower(b.src_id) like 'ehr site%'
-                UNION DISTINCT
-                SELECT DISTINCT person_id
-                FROM \`$BQ_PROJECT.$BQ_DATASET.visit_occurrence\` as a
-                LEFT JOIN \`$BQ_PROJECT.$BQ_DATASET.visit_occurrence_ext\` as b on a.visit_occurrence_id = b.visit_occurrence_id
-                WHERE lower(b.src_id) like 'ehr site%'
-            ) b on a.person_id = b.person_id
-    ) y
-WHERE x.person_id = y.person_id"
-
+if [ "$BQ_PROJECT:$BQ_DATASET" == "all-of-us-ehr-dev:synthetic_cdr20180606" ]
```

Comment:
Again Synth dataset doesn't have any of the OMOP *_ext tables. This is how we determine if data is EHR or not